### ISSUE TITLE
oe_get_seal_key_by_policy_v2 with SGX does not populate key_id

### ISSUE THREAD

**Issue by the729** [openenclave/openenclave#4665]
I found the SGX implementation of `oe_get_seal_key_by_policy_v2` does not populate the random `key_id` field before calling `EGETKEY`. 
https://github.com/openenclave/openenclave/blob/df8cb0d1807e80e5f9148dcad3841050a78ed6ad/enclave/core/sgx/keys.c#L300

Is this an expected behavior? If so, how are we expected to use this function securely?

**Comment by the729** [comment]
@mingweishih Thank you for your quick response. 

I have checked the sealing example. Despite the README doc saying 'OE decides to only provide generic routine for getting seal key...', the example code actually uses `oe_seal` and `oe_unseal`, not `oe_get_seal_key_by_policy` and `oe_get_seal_key` stated in the doc.

And the random key_id is initialized inside `oe_seal`.

https://github.com/openenclave/openenclave/blob/df8cb0d1807e80e5f9148dcad3841050a78ed6ad/samples/data-sealing/common/dispatcher.cpp#L27
https://github.com/openenclave/openenclave/blob/df8cb0d1807e80e5f9148dcad3841050a78ed6ad/samples/data-sealing/common/dispatcher.cpp#L96


Does this mean `oe_get_seal_key_by_policy` and `oe_get_seal_key` are no longer recommended? Or even deprecated?

**Comment by the729** [comment]
After a deeper dive, I found that `oe_get_seal_key_by_policy` does not use KeyID since its initial implementation. 
So before `oe_seal` and `oe_unseal` was implemented, the recommended usage of `oe_get_seal_key_by_policy` was actually not following the recommendations in Intel(R) SGX Developer Guide (screenshot as follow). 

<img width="789" alt="image" src="https://user-images.githubusercontent.com/8331317/203513861-fcd45792-2484-4a8d-9d44-967efb63962d.png">

Should this be fixed?

**Comment by the729** [comment]
Ego project uses `oe_get_seal_key_by_policy`. I have also filed an issue there:
https://github.com/edgelesssys/ego/issues/189

**Comment by thomasten** [comment]
Yes, we need to get the key directly for EGo and also want to continue doing it like this in our other products.

**Comment by anakrish** [comment]
@mingweishih Should `oe_get_seal_key_by_policy` be deprecated?

**Comment by mingweishih** [comment]
Based on @thomasten's comment, we might need to fix the API. But from OE's perspective, the API is treated as deprecated, the recommendation is using `oe_seal`.
